jina/types/arrays/document.py

```diff
@@ -445,3 +446,16 @@ def load_binary(cls, file: Union[str, BinaryIO]) -> 'DocumentArray':
             dap.ParseFromString(fp.read())
             da = DocumentArray(dap.docs)
             return da
+
+    def sample(self, k: int) -> 'DocumentArray':
+        """random sample k elements from :class:`DocumentArray` without replacement
+
+        :param k: Number of elements to sample.
+        """
+        if k < len(self):
+            raise ValueError('DocumentArray do not have enough Document to sample')
+        indices = random.sample(range(len(self)), k)
+        sampled = DocumentArray()
```

Comment:
I think u can do `sampled = self[indices]`

Reply:
it's not woking

Reply:
why not? we should maybe understand why first?

Reply:
@JoanFM it's not a python syntax, it's supported by numpy only